### PR TITLE
Document hashlib's API

### PR DESCRIPTION
**Pull request by lmctv**
Add some docs about scrypt and blake2b as implemented in nacl.hashlib as requested in #248

### CODE REVIEW

**Comment by reaperhulk** [comment]
I think there's an instance of `pw_hash` in the docs that needs to be updated as well?

File: docs/api/hashlib.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+nacl.hashlib
+============
+
+The :py:mod:`nacl.hashlib` module exposes directly usable implementations
+of raw constructs which libsodium exposes with simplified APIs, like the
+ones in :py:mod:`nacl.hash` and in :py:mod:`nacl.pw_hash`.
```
Comment (reaperhulk) [inline docs/api/hashlib.rst]: This needs to be pwhash now

File: docs/api/hashlib.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+When such constructs are exposed in some cpython version standard library,
```
Comment (reaperhulk) [inline docs/api/hashlib.rst]: I don't think this paragraph is what we want to say. The current implementations are compatible with the standard library is all we really need to say. The stdlib is extraordinarily unlikely to break backwards compatibility

**Comment by lmctv** [comment]
@reaperhulk  I think this PR is ready for another review cycle. Since I'd grown unhappy with discussing scrypt before blake2b, I swapped them in my last commit; if the original order seems better... I't just a single  commit to trim before merging.

**Pull request merged**